lib/stanford-mods/imprint.rb

```diff
@@ -37,8 +37,44 @@ def display_str
         imprint_statements.join('; ') if imprint_statements.present?
       end
 
+      def publication_date_for_slider
+        @originInfo_ng_nodeset.map do |origin_info_node|
+
+          date_elements = if origin_info_node.as_object.first.key_dates.any?
+                            origin_info_node.as_object.first.key_dates.map(&:as_object).map(&:first)
+                          else
+                            date_field_keys.map do |date_field|
+                              next unless origin_info_node.respond_to?(date_field)
+                              date_elements = origin_info_node.send(date_field)
+                              date_elements.map(&:as_object).map(&:first) if date_elements.any?
+                            end.first
+                          end
+
+          if date_elements.none?
+            []
+          elsif date_elements.find(&:start?) && date_elements.find(&:end?)
+            start_date = date_elements.find(&:start?)
+            end_date = date_elements.find(&:end?)
+
+            (start_date.as_range.min.year..end_date.as_range.max.year).to_a
+          elsif date_elements.find(&:start?)
```

Comment:
can you `elsif start_date = date_elements.find(&:start?)` here?

Reply:
I could, but does it help readability? The list of date elements (oughn't..) be very long; for the most part, it's 1-3 items, so performance isn't a huge concern.

Reply:
Okay, and we're not going to run this code on tens of thousands of records at a time?

Reply:
No, this code acts on single records.